corehq/apps/app_manager/xform.py

```diff
@@ -1511,7 +1571,8 @@ def check_case_type(action):
                     update_case_block.add_close_block(self.action_relevance(action.close_condition))
 
                 if has_schedule:
-                    configure_visit_schedule_updates(update_case_block.update_block)
+                    self.add_casedb()
+                    configure_visit_schedule_updates(update_case_block.update_block, action)
```

Comment:
Edge case but I think you should pass `session_case_id` into this function rather than getting it from the action since the action may not have the correct value (in the case of submodules).